### PR TITLE
Fix: Update layout structure and minor tweaks

### LINKED ISSUE #208: Docs page has no title
If I go here, the webpage has no title:

https://sonarwhal.com/docs/

`<!doctype html><html lang="en"><head><meta charset="utf-8"><title></title><meta name="description" content="sonar website">`

### LINKED ISSUE #216: Add multiple layouts
Right now we only have the `index` layout with lots of conditions depending on categories and names. We should split this into multiple different layouts and create partials for the common parts. This will make adding new things like #212 easier to add.

### PR DESCRIPTION
**Pull request by molant**
* Updated the layout structure so adding new pages with different
  content is easier from now on.
* Modified search page to handle the no results case
* Other minor tweaks

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Fix #208
Fix #216

### CODE REVIEW

**Comment by molant** [comment]
I currently have a problem with the `/docs` page. Working on it.

**Comment by molant** [comment]
Will be nice if when you are testing you can check each individual page. I've done it locally but there are many changes.

Hopefully this should make adding new pages with custom content easier (like #212).

@ststimac, I have an issue with the `/docs` page:

image

Guides boxes aren't aligned with the search box. I guess I forgot to add something? Can you please take a look? Also the search icon is broken on Edge (it also happens in production 😥).

Thanks!

File: src/hexo/themes/documentation/layout/partials/styles.hbs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+<link rel="stylesheet" href="{{url_for "/core/css/media.css "}}">
+<link rel="stylesheet" href="{{url_for "/core/css/navigation.css "}}">
+<link rel="stylesheet" href="{{url_for "/core/css/search-results.css "}}">
+<link rel="stylesheet" href="{{url_for "/core/css/structure.css "}}">
+<link rel="stylesheet" href="{{url_for "/core/css/type.css "}}">
+<link rel="stylesheet" href="{{url_for "/core/css/highlight.css "}}">
+<link rel="stylesheet" href="{{url_for "/components/breadcrumb/breadcrumb.css "}}">
+<link rel="stylesheet" href="{{url_for "/components/pagination/pagination.css "}}">
+<link rel="stylesheet" href="{{url_for "/core/css/controls.css "}}">
+<link href="https://fonts.googleapis.com/css?family=Montserrat:200,300,400,500,600" rel="stylesheet">
```
Comment (alrra) [inline src/hexo/themes/documentation/layout/partials/styles.hbs]: Do we need all those sizes?

Comment (molant) [reply src/hexo/themes/documentation/layout/partials/styles.hbs]: @alrra I have no idea. I just moved things from one place to another. @ststimac might know more but probably worth opening an issue

Comment (alrra) [reply src/hexo/themes/documentation/layout/partials/styles.hbs]: [Done](https://github.com/sonarwhal/sonarwhal.com/issues/218).

Comment (ststimac) [reply src/hexo/themes/documentation/layout/partials/styles.hbs]: @alrra I think we are using all of these, but will double check.

**Comment by ststimac** [comment]
@molant I'll add an issue for the search icon & margin issue for me to look into

**Pull request closed without merging**